packages/core/core/src/index.ts

```diff
@@ -10,11 +10,18 @@ export * from './plugin';
 export { annotationUtils, Annotation };
 
 export abstract class Type {
-  abstract deriveLiteral(
+  public abstract deriveLiteral(
     parentAnnotations: Annotation[],
     context: MantaStyleContext,
   ): Promise<Type>;
   public mock(): any {
     throw new Error('Literal types should be derived before mock.');
   }
 }
+
+export abstract class CustomType extends Type {
+  public abstract typeForAssignabilityTest(
```

Comment:
`CustomType` is literally designed for assignability test. Say we have a custom type `ShopeeAvatar` which extends `Type`. We know it should be typed as `string` as compile-time. However, in `isAssignable` function, the way to test if a type is `string` is `type instanceof StringKeyword`, which is not the case for `ShopeeAvatar`. Therefore we should have a callback to get the type it should be typed as, `typeForAssignablityTest`. It is not necessary to be the same as `deriveLiteral` because sometimes you don't have to make the actual HTTP request here but you do in `deriveLiteral`. Another case is `Delay` in `helpers`, it has a certain amount of delay while `deriveLiteral` but it doesn't in `typeForAssignablityTest`.